x/programs/cmd/simulator/test/context_injection/src/lib.rs

```diff
@@ -112,7 +112,7 @@ mod tests {
             .unwrap();
 
         let mut test_context = TestContext::from(program_id);
-        test_context.actor_key = Some(actor_key);
+        test_context.actor = Address::new([1; Address::LEN]);
```

Comment:
I actually tried to implement an `Address::random` function. That didn't worked, at all...
1. There is no way to have the `random` function being `#[cfg(test)]` and using it across crates (typically example, since they are importing the simulator)
2. No way to have it using a custom `test-utils` flag because then the WASM target will complain that it has the `rand` crate.
It's probably okay this way for now.

Reply:
I have instead created an `Address::from_str` that takes in a name and convert it to bytes behind the `test-utils` feature. Coming in the next PR!

Reply:
Why don't you just return an address when you create a key. What you're doing here doesn't really make sense and I don't quite understand your explanation. 

>I have instead created an Address::from_str that takes in a name and convert it to bytes behind the test-utils feature. Coming in the next PR!

I don't quite understand this either.

Wouldn't it be easier if you just passed in an `Address` to create_key?

Reply:
The `create_key` endpoint was meant for manual interaction with the simulator. It was just constraining for no reason when interacting with it programmatically, because we had to store named keys.
In the simulator, we don't really need private components, it's much easier if everything can be impersonated by just injecting an address to the `testContext`.

Reply:
I would leave private components to the vm